### PR TITLE
Project Memberships

### LINKED ISSUE #135: Authors Automatically Follow their Project
Just noticed, if you author a project you have to click follow to follow it. Not sure what the functionality is intended to be, but it might be good to have author's automatically follow their own project.

### PR DESCRIPTION
**Pull request by brec9824**
Added a member page to projects which requires access from admin of the project.
Features:
- Request access to project
- Admin add user
- Admin deny user
- Editable banner, mission statement, custom tab on project member page
- Kicking users from project

Also fixes #135, #127, #114, #99, #12, #11, #10 (Deleting account I setup some placeholder profile routes and controller)

Ready for review @uidaho/lambda 

Notes: added chat to the bottom right hand side of the project members page

### CODE REVIEW

**Comment by brec9824** [comment]
Ok @backburn @rickbau5 fixed the database migration so it adds to old tables rather than editing them. Ready for review.

**Comment by brandonjank** [comment]
https://squireproject.com/project/Project%20One
Join Project and Follow buttons show up when not logged in, but do not show up when logged in.
Really, all buttons should be available or context aware. If you don't have permission it should ask you to login etc...

**Comment by brec9824** [comment]
@backburn Join project and follow buttons show when not logged in or not the owner of the project. If you are the owner of the project the those two buttons are hidden since the owner doesnt need them.

**Comment by brandonjank** [comment]
im not the owner of the projkect, project->user_id for that project is 0.

Im thinking you are relying on a record being present in the db in all the project_ tables for each user?

**Comment by brandonjank** [comment]
In this case, if there is no row in the table for the user, it should create one. Check out: firstOrCreate() and firstOrNew()

**Comment by brec9824** [comment]
@backburn Im a little confused how to replicate what you did. Can you tell me what you did to have that happen. Because I tried nuking my database and seeding it with some projects. Then registering a account and checking one of the projects and the buttons were there.

**Comment by brandonjank** [comment]
my account was already there, php artisan migrate worked fine. So, there are no records for my user id in the project_members or project_followers. I believe you are depending on there being an record in there, it should try or create instead, so this does not happen.

**Comment by brec9824** [comment]
@backburn Ok gotcha. I see a problem this will cause without wiping everyone's database. All previous projects will become inaccessible to anyone because all projects become connected to a user through memberships. I would have to detect all users with projects created and create memberships for them also.

File: app/Http/Controllers/ProjectMembersController.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+     * @param Project $project
+     * @param User $user
+     * @return current view
+     */
+    public function denyMembershipRequest(Project $project, User $user)
+    {
+        $this->authorize('authToAddDeny', [$project, $user->id]);
+
+        $project->deleteMembershipRequest($user->id);
+        //Session::flash('membership_request_success', 'You are no longer requesting membership to this project.');
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectMembersController.php]: comments in this file?

**Comment by rickbau5** [comment]
Looks good to merge to me. I'm fine with this nuking the databases if need be. It is a dev environment after!

**Comment by brec9824** [comment]
Ok will merge after I fix one thing I just noticed.

**Comment by brec9824** [comment]
Just so everyone knows you will have nuke your database then remigrate and also might require doing a composer update.

**Pull request merged**